components/support/payload/src/lib.rs

```diff
@@ -0,0 +1,177 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use serde::Serialize;
+use std::{
+    io::{self, Write},
+    num::NonZeroUsize,
+};
+
+/// A quantity of items that can fit into a payload, accounting for
+/// serialization, encryption, and Base64-encoding overhead.
+pub enum Fit {
+    /// All items in the list can fit into the payload.
+    All,
+
+    /// Some, but not all, items can fit into the payload. The list should be
+    /// truncated to this number of elements, to avoid exceeding the
+    /// maximum payload size.
+    Some(NonZeroUsize),
+
+    /// The maximum payload size is too small to hold any items in the list.
+    None,
+
+    /// The serialized size of the items couldn't be determined because of
+    /// an error.
+    Err(serde_json::Error),
+}
+
+impl Fit {
+    #[inline]
+    pub fn truncate_to(&self) -> Option<NonZeroUsize> {
```

Comment:
`truncate_to` seems like it's actually taking action, but really isn't returning an indicator that the caller should take action. I wonder if `needs_truncation` might be better? I really don't care though!